Tests/tests/test_ScreenCapture.cpp

```diff
@@ -0,0 +1,192 @@
+protected:
+    Core::ProxyType<Plugin::ScreenCapture> plugin;
+    Core::JSONRPC::Handler& handler;
+    Core::JSONRPC::Connection connection;
+    string response;
+
+    ScreenCaptureTest()
+        : plugin(Core::ProxyType<Plugin::ScreenCapture>::Create())
+        , handler(*(plugin))
+        , connection(1, 0)
+    {
+    }
+    virtual ~ScreenCaptureTest() = default;
+};
+
+class ScreenCaptureEventTest : public ScreenCaptureTest {
+protected:
+    ServiceMock service;
+    Core::JSONRPC::Message message;
+    FactoriesImplementation factoriesImplementation;
+    PluginHost::IDispatcher* dispatcher;
+
+    ScreenCaptureEventTest()
+        : ScreenCaptureTest()
+    {
+        EXPECT_EQ(string(""), plugin->Initialize(nullptr));
+
+        PluginHost::IFactories::Assign(&factoriesImplementation);
+
+        dispatcher = static_cast<PluginHost::IDispatcher*>(
+            plugin->QueryInterface(PluginHost::IDispatcher::ID));
+        dispatcher->Activate(&service);
+    }
+    virtual ~ScreenCaptureEventTest() override
+    {
+        dispatcher->Deactivate();
+        dispatcher->Release();
+
+        PluginHost::IFactories::Assign(nullptr);
+
+        plugin->Deinitialize(nullptr);
+    }
+};
+
+class ScreenCaptureFrameBufferTest : public ScreenCaptureEventTest {
+protected:
+    FrameBufferApiImplMock frameBufferApiImplMock;
+
+    ScreenCaptureFrameBufferTest()
+        : ScreenCaptureEventTest()
+    {
+        FrameBufferApi::getInstance().impl = &frameBufferApiImplMock;
+    }
+    virtual ~ScreenCaptureFrameBufferTest() override
+    {
+        FrameBufferApi::getInstance().impl = nullptr;
+    }
+};
+
+TEST_F(ScreenCaptureTest, RegisteredMethods)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("uploadScreenCapture")));
+}
+
+TEST_F(ScreenCaptureFrameBufferTest, FrameBufferUpload)
+{
+    FBContext fbcontext;
+    PixelFormat pixelFormat = {32, 24, 0, 1, 255, 255, 255, 16, 8, 0};
+    vnc_uint8_t* frameBuffer = (vnc_uint8_t*) malloc(5120 * 720);
+    memset(frameBuffer, 0xff, 5120 * 720);
+    
+    Core::Event uploadComplete(false, true);
+    
+    EXPECT_CALL(frameBufferApiImplMock, fbCreate(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext** fbctx) {
+                *fbctx = &fbcontext;
+                return ErrNone;
+            }));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbInit(::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext* fbctx, VncServerFramebufferAPI* server, void* serverctx) {
+                if (nullptr != server)
+                {
+                    server->framebufferUpdateReady(&fbcontext);
+                    server->framebufferDetailsChanged(&fbcontext, nullptr, 1280, 720, 5120, nullptr);
+                    server->paletteChanged(&fbcontext, nullptr);
+                    server->logMsg(&fbcontext, "msg");
+                }
```

Comment:
as a reader, I don't understand what these lines are for, even after checking the code. is it necessary?

Reply:
if these lines don't verify anything please remove them to improve readability

Reply:
Removed.